src/Cli/dotnet/CommandLineInfo.cs

```diff
@@ -55,4 +58,101 @@ private static string GetDisplayRid(DotnetVersionFile versionFile)
             currentRid :
             versionFile.BuildRid;
     }
+
+    public static void PrintCliSchema(Command command)
+    {
+        var options = new JsonWriterOptions { Indented = true };
+        using var stream = new MemoryStream();
+        using var writer = new Utf8JsonWriter(stream, options);
+
+        writer.WriteStartObject();
+        TraverseCli(command, writer);
+        writer.WriteEndObject();
+        writer.Flush();
+
+        string json = Encoding.UTF8.GetString(stream.ToArray());
+        Console.WriteLine(json);
+    }
+
+    private static void TraverseCli(Command command, Utf8JsonWriter writer)
+    {
+        writer.WriteString(nameof(command.Description).ToCamelCase(), command.Description);
+        writer.WriteBoolean(nameof(command.Hidden).ToCamelCase(), command.Hidden);
+        writer.WriteBoolean("hasValidators", command.GetHasValidators() ?? false);
+
+        writer.WriteStartArray(nameof(command.Aliases).ToCamelCase());
+        foreach (var alias in command.Aliases.Order())
+        {
+            writer.WriteStringValue(alias);
+        }
+        writer.WriteEndArray();
+
+        writer.WriteBoolean(nameof(command.TreatUnmatchedTokensAsErrors).ToCamelCase(), command.TreatUnmatchedTokensAsErrors);
+
+        writer.WriteStartObject(nameof(command.Arguments).ToCamelCase());
+        foreach (var argument in command.Arguments.OrderBy(a => a.Name))
+        {
+            // TODO: Check these
+            writer.WriteStartObject(argument.Name);
+
+            writer.WriteString(nameof(argument.Description).ToCamelCase(), argument.Description);
+            writer.WriteBoolean(nameof(argument.Hidden).ToCamelCase(), argument.Hidden);
+            writer.WriteBoolean("hasValidators", argument.GetHasValidators() ?? false);
+            writer.WriteString(nameof(argument.HelpName).ToCamelCase(), argument.HelpName);
+            writer.WriteString(nameof(argument.ValueType).ToCamelCase(), argument.ValueType.FullName);
+            writer.WriteBoolean(nameof(argument.HasDefaultValue).ToCamelCase(), argument.HasDefaultValue);
+
+            writer.WriteStartObject(nameof(argument.Arity).ToCamelCase());
+            writer.WriteNumber(nameof(argument.Arity.MinimumNumberOfValues).ToCamelCase(), argument.Arity.MinimumNumberOfValues);
+            writer.WriteNumber(nameof(argument.Arity.MaximumNumberOfValues).ToCamelCase(), argument.Arity.MaximumNumberOfValues);
+            writer.WriteEndObject();
+
+            writer.WriteEndObject();
+        }
+        writer.WriteEndObject();
```

Comment:
TODO: Local functions might make sense for each of these sub-sections.

Reply:
I've already moved this information into a new file and made separate methods for each.